spec/helpers.lua

```diff
@@ -1709,6 +1712,60 @@ local function wait_for_all_config_update(timeout, admin_client_timeout, forced_
 end
 
 
+--- Waits for a file to meet a certain condition
+-- The check function will repeatedly be called (with a fixed interval), until
+-- there is no Lua error occurred
+--
+-- NOTE: this is a regular Lua function, not a Luassert assertion.
+-- @function wait_for_file_status
+-- @tparam string status "created" or "deleted" or "socket"
+-- 
+-- created: wait for file to be created
+-- 
+-- deleted: wait for file to be deleted
+-- 
+-- socket:  wait for file to become a socket file
+-- 
+-- @tparam string path the file path
+-- @tparam[opt=10] number timeout maximum time to wait
+local function wait_for_file_status(status, path, timeout)
```

Comment:
consider use `stat` call, you can use `lfs.attribute(path)` (https://lunarmodules.github.io/luafilesystem/manual.html#introduction). and let's not make `socket` a parrelle status as `created`, instead we can say 
`wait_for_file(mode, path, timeout)` then we can further simplify this function .

Reply:
done